### PR TITLE
Add snapshot module allowing to restore state

### LINKED ISSUE #101: Add helpers for snapshots
@jamesmorgan suggested on the Telegram channel adding support for `evm_snapshot` and `evm_revert`. Given that their semantics are a bit strange, and that a loot of speed can be gained with proper usage of snapshots, I think this is a great idea.

I'm not sure on the API, we could definitely simply provide simple functions to call these, but it might be possible to achieve a more ergonomic solution. @jamesmorgan could you share more about your use case to see if we can tailor our solution to it? Thanks!

### PR DESCRIPTION
**Pull request by pawurb**
Hi. This feature was mentioned in [#101] but abandoned. This PR adds a `snapshot` method which allows to restore time. 

Usage in tests could look like:

```
let snapshot;
beforeEach(async() => {
    snapshot = await time.snapshot();
});
afterEach(async() => {
    await snapshot.restore();
});
```

Fixes #101

### CODE REVIEW

**Comment by frangio** [comment]
I think this helper makes sense, although not in the `time` module since snapshots are about much more than just time. I don't think there's a module where this would fit, so perhaps it should be its own top level export.

**Comment by pawurb** [comment]
a `snapshot` module exporting a single `snaphot` function would work for you?

**Comment by frangio** [comment]
The `snapshot` module should be the function itself, kind of like `expectRevert`.

https://github.com/OpenZeppelin/openzeppelin-test-helpers/blob/ccd8d770c5daf6087bb487ccf35cbacae5af60ad/src/expectRevert.js#L82

**Comment by pawurb** [comment]
@frangio fixed. Let me know if looks ok. I can add some docs unless you prefer to do it.

File: test/src/snapshot.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const { BN } = require('../../src/setup');
+const { expect } = require('chai');
+
+const snapshot = require('../../src/snapshot');
+const time = require('../../src/time');
+
+describe('snapshot', function () {
```
Comment (frangio) [inline test/src/snapshot.test.js:7]: Can you add a test where you snapshot, then send some ether to another account, then restore the snapshot and confirm that the ether transfer was rolled back?

**Review by frangio — COMMENTED** [review]
This is great, thanks!

Can you add some brief documentation in `docs/modules/ROOT/pages/api.adoc`. You should copy the style from the other modules.

**Comment by pawurb** [comment]
@frangio hope it's ok now.

**Review by frangio — APPROVED** [review]
Awesome work, thanks!

For future PRs with us please avoid rebasing and just add commits on top, it makes reviews a little easier. :slightly_smiling_face:

**Pull request merged**